docs/faq/programming.md

```diff
@@ -0,0 +1,176 @@
+| 63 | See `optional(Type).get()`. |
+| 64 | `tvm.buildExtMSg()` call with wrong parameters. `See tvm.buildExtMsg()`. |
+| 66 | Convert an integer to a string with width less than number length. See `format()`.
+| 67 | See `gasToValue` and `valueToGas`. |
+| 68 | There is no config parameter 20 or 21. |
+| 69 | Zero to the power of zero calculation (0**0 in solidity style or 0^0). |
+| 70 | string method substr was called with substr longer than the whole string. |
+| 71 | Function marked by externalMsg was called by internal message. |
+| 72 | Function marked by internalMsg was called by external message. |
+| 73 | The value can't be converted to enum type. |
+| 74 | Await answer message has wrong source address. |
+| 75 | Await answer message has wrong function id. |
+| 76 | Public function was called before constructor. |
+| 77 | It's impossible to convert variant type to target type. See `variant.toUint()`. |
+| 78 | There's no private function with the function id. |
+| 79 | You are deploying contract that uses pragma upgrade func/oldsol. Use the contract only for updating another contracts. |
+
+Any error with a code over 100 is a user-defined error. Check the code in the contract you call.
+
+You can check more information about errors in threaded solidity [specification](https://github.com/tonlabs/TON-Solidity-Compiler/blob/master/API.md#tvm-exception-codes)
+
+</details>
+
+<details>
+<summary>
+My contract needs to receive fungible tokens. How can I do it?
+</summary>
+
+Your contract should have a TIP3 TokenWallet to perform this action. It is not necessary how exactly your contract will get its wallet (ex. you can call [deployWallet](../standards/TIP-3/2.md#deploy-token-wallet) of TokenRoot inside your contract constructor or deploy it by yourself). Next, your contract should implement an [onAcceptTokensTransfer](../standards/TIP-3/2.md#incoming-transfer-callback) callback. Pay attention, that a sender should set [notify](../standards/TIP-3/2.md#transfer-tokens-to-the-recipient) flag to true for this callback to be called.
+
+Check out [this](../build/development-guides/how-to-create-your-own-non-fungible-tip-4-token/venom-in-action/simple-nft-auction.md) guide, which shows how to implement a simple auction contract that accepts TIP-3 tokens.
+
+Of course, you need to check [TIP-3 standards](../standards/TIP-3/core-description.md) too.
+
+</details>
+
+<details>
+<summary>
+There are amount of fees including storage fees. How should I manipulate gas to be sure that my contract will be able to pay for all of them?
+</summary>
+
+You need to think out the gas management. You need to manipulate with flags of the message your contract sends and use [`tvm.rawReserve()`](https://github.com/tonlabs/TON-Solidity-Compiler/blob/master/API.md#tvmrawreserve) function. 
+
+Check out [this](../build/development-guides/how-to-create-your-own-fungible-tip-3-token/venom-in-action/simple-tokensale.md) guide, that raises the issue of gas management.
```

Comment:
Check out [this](../build/development-guides/how-to-create-your-own-fungible-tip-3-token/venom-in-action/simple-tokensale.md) guide, here we explore the issue of gas management.